sync/src/synchronizer/mod.rs

```diff
@@ -106,20 +108,23 @@ impl BlockFetchCMD {
     }
 
     fn send_getblocks(v_fetch: Vec<packed::Byte32>, nc: &ServiceControl, peer: PeerIndex) {
+        debug!("send_getblocks len={:?} to peer={}", v_fetch.len(), peer);
+
         let content = packed::GetBlocks::new_builder()
-            .block_hashes(v_fetch.clone().pack())
+            .block_hashes(v_fetch.pack())
             .build();
         let message = packed::SyncMessage::new_builder().set(content).build();
-
-        debug!("send_getblocks len={:?} to peer={}", v_fetch.len(), peer);
         if let Err(err) = nc.send_message_to(
             peer,
             SupportProtocols::Sync.protocol_id(),
             message.as_bytes(),
         ) {
-            debug!("synchronizer send GetBlocks error: {:?}", err);
+            debug!("synchronizer send_getblocks error: {:?}", err);
         }
-        crate::synchronizer::metrics_counter_send(message.to_enum().item_name());
+
+        let bytes = message.as_bytes().len() as u64;
+        metrics!(counter, "ckb.messages_total", 1, "direction" => "out", "name" => "GetBlocks");
```

Comment:
if `messages_total` always being `1`, why we need this metric?

Reply:
It's a counter. A global counter named "ckb.messages_total" will increase by 1 when call `metrics!(counter, "ckb.messages_total", 1, "direction" => "out", "name" => "GetBlocks");`